examples/commit_batch/src/lib.rs

```diff
@@ -0,0 +1,141 @@
+use anyhow::Result;
+use fxhash::FxHashMap;
+use nomt::{KeyPath, KeyReadWrite, Node, Nomt, Options, Session, Witness, WitnessedOperations};
+use sha2::Digest;
+use std::{collections::hash_map::Entry, path::PathBuf};
+
+const NOMT_DB_FOLDER: &str = "nomt_db";
+
+pub struct NomtDB;
+
+impl NomtDB {
+    pub fn commit_batch() -> Result<(Node, Node, Witness, WitnessedOperations)> {
+        // Define the options used to open NOMT
+        let opts = Options {
+            path: PathBuf::from(NOMT_DB_FOLDER),
+            fetch_concurrency: 1,
+            traversal_concurrency: 1,
+        };
+
+        // Open NOMT database, it will create the folder if it does not exits
+        let nomt = Nomt::open(opts)?;
+
+        // Create a new Session object
+        //
+        // During a session, the backend is responsible for returning read keys
+        // and receiving hints about future writes
+        //
+        // Writes do not occur immediately, instead,
+        // they are cached and applied all at once later on
+        let mut session = nomt.begin_session();
+
+        // This access map is used to track operations on the backend during a session
+        // `KeyReadWrite` abstracts actions performed on NOMT
+        let mut access: FxHashMap<KeyPath, KeyReadWrite> = FxHashMap::default();
```

Comment:
I think this is overkill, as well as the `read` function. That's creating higher-level logic on top of NOMT - we are using an `FxHashMap` in testing, but it's not the only way.

NOMT just expects `commit` to receive a sorted set of operations.